docs/content/best-practices/validation.md

```diff
@@ -0,0 +1,34 @@
+---
+title: "Validation"
+weight: 50
+---
+
+# Validation
+
+There are a number of ways to add client-side validation to resources. The benefit of client-side validation is that errors can be surfaced at plan time, instead of partway through a (potentially very long) apply process, allowing for faster iteration. However, the tradeoff is that client-side validation can get out of sync with server-side validation, creating additional maintenance burden for the provider and preventing users from accessing the latest features without upgrading.
+
+Client-side validation is generally discouraged due to the low positive impact of an individual validation rule and outsized negative impact when client-side validation and API capabilities drift, requiring both provider changes and users to update. Client-side validation may be added in cases where it is extremely unlikely to change, covered below.
+
+The following sections cover best practices for specific types of client-side validation.
+
+## URL segments
+
+If a resource URL looks like:
+
+```
+projects/{project}/folders/{folder}/resource/{resource_id}
+```
+
+Adding validation for the last part of the path (`resource_id`) may be safe if there are specific restrictions that are not going to change. However, if the API was ever less restrictive (or becomes less restrictive later), resources created with other tools may not be possible to import into Terraform safely.
```

Comment:
Can you elaborate on what would be _unsafe_ specifically? I think that users would not be successful because we would reject their config and/or fail on import if we had apply time assumptions that were wrong.

Reply:
Done.